### PR TITLE
MSP-2748 Allow NodePort configuration

### PR DESCRIPTION
**Pull request by dstaroff**

### CODE REVIEW

File: helm/slurm-cluster/templates/slurm-cluster-cr.yaml [hunk 1]
```diff
@@ -128,12 +128,13 @@ spec:
       sshRootPublicKeys:
         {{- default list .Values.slurmNodes.login.sshRootPublicKeys | toYaml | nindent 8 }}
       sshdServiceType: {{ default "LoadBalancer" .Values.slurmNodes.login.sshdServiceType | quote }}
-      sshdServiceAnnotations:
-        external-dns.alpha.kubernetes.io/hostname: "{{ include "slurm-cluster.name" . }}.slurm-operator.msp-dev.nemax.nebius.cloud."
+      sshdServiceAnnotations: {}
+      {{- /*external-dns.alpha.kubernetes.io/hostname: "{{ include "slurm-cluster.name" . }}.slurm-operator.msp-dev.nemax.nebius.cloud."*/ -}}
```
Comment (rdjjke) [inline helm/slurm-cluster/templates/slurm-cluster-cr.yaml:132]: Why is it commented? Should it be conditional?

Comment (dstaroff) [reply helm/slurm-cluster/templates/slurm-cluster-cr.yaml:132]: Right now there is no possibility to create DNS records in Newbius. Especially, via service annotations.
We will tidy this up when it's possible

File: api/v1/slurmcluster_types.go [hunk 2]
```diff
@@ -405,6 +406,12 @@ type SlurmNodeLogin struct {
 	// +kubebuilder:validation:Optional
 	SshdServiceLoadBalancerIP string `json:"sshdServiceLoadBalancerIP,omitempty"`
 
+	// SshdServiceNodePort represents the port to be opened on nodes in case of NodePort type of service
+	//
+	// +kubebuilder:validation:Optional
+	// +kubebuilder:default=0
```
Comment (asteny) [inline api/v1/slurmcluster_types.go:412]: Why port is 0? What does it mean?

Comment (dstaroff) [reply api/v1/slurmcluster_types.go:412]: It's used in validation to be sure it was set if service type is `NodePort`

File: helm/slurm-cluster/templates/slurm-cluster-cr.yaml (hunk 1 above)
Comment (rdjjke) [reply helm/slurm-cluster/templates/slurm-cluster-cr.yaml:132]: Ah, OK, got it

**Pull request merged**